.goreleaser.yaml

```diff
@@ -0,0 +1,287 @@
+# Documentation at https://goreleaser.com
+version: 2
+project_name: editorconfig-checker
+env:
+  - REPO_LICENSE=MIT
+  - DOCKERHUB_USERNAME={{ if index .Env "DOCKERHUB_USERNAME"  }}{{ .Env.DOCKERHUB_USERNAME }}{{ else }}echoix{{ end }}
+  - GHCR_USERNAME={{ if index .Env "GHCR_USERNAME"  }}{{ .Env.GHCR_USERNAME }}{{ else }}echoix{{ end }}
+before:
+  hooks:
+    - echo {{.Version}} > VERSION
+    - go mod tidy
+    - go generate ./...
+builds:
+  - id: "editorconfig-checker"
+    main: ./cmd/editorconfig-checker
+    binary: editorconfig-checker
+    env:
+      - CGO_ENABLED=0
+    goos:
+      - linux
+      - windows
+      - darwin
+
+    goarch:
+      - amd64
+      - arm64
+      - arm
+      - '386'
+    mod_timestamp: "{{ .CommitTimestamp }}"
+    flags:
+      - -trimpath
+    ldflags:
+      - -s -w -X main.version={{.Version}}
+      - -X main.commit={{.Commit}} -X main.date={{.CommitDate}}
+      - -X main.builtBy=goreleaser
+
+  # this build is used to provide the legacy-style archives
+  - id: "ec-compat"
+    main: ./cmd/editorconfig-checker
+    binary: bin/ec-{{- .Os }}-{{ .Arch }}{{- if .Arm }}v{{ .Arm }}{{ end }}
+    env:
+      - CGO_ENABLED=0
+    goos:
+      - linux
+      - windows
+      - darwin
+
+    goarch:
+      - amd64
+      - arm64
+      - arm
+      - '386'
+    mod_timestamp: "{{ .CommitTimestamp }}"
+    flags:
+      - -trimpath
+    ldflags:
+      - -s -w -X main.version={{.Version}}
+      - -X main.commit={{.Commit}} -X main.date={{.CommitDate}}
+      - -X main.builtBy=goreleaser
+
+
+universal_binaries:
+  - replace: true
+archives:
+  - id: archive-editorconfig-checker
+    builds:
+      - editorconfig-checker
+
+    format: tar.gz
+    name_template: >-
+      {{ .ProjectName }}-{{- .Os }}-{{ .Arch }}{{- if .Arm }}v{{ .Arm }}{{ end }}
+    format_overrides:
+      - goos: windows
+        format: zip
+  - id: archive-ec
+    builds:
+      - ec-compat
+
+    format: tar.gz
+    name_template: >-
+      ec-{{- .Os }}-{{ .Arch }}{{- if .Arm }}v{{ .Arm }}{{ end }}
+    format_overrides:
+      - goos: windows
+        format: zip
+
+nfpms:
+  - vendor: Editorconfig-Checker
+    homepage: https://editorconfig-checker.github.io/
+    maintainer: Max Strübing <mxstrbng@gmail.com>
+    description: |-
+      A tool to verify that your files are in harmony with your .editorconfig
+    license: "{{ .Env.REPO_LICENSE }}"
+    formats:
+      - apk
+      - deb
+      - rpm
+      - termux.deb # Since GoReleaser v1.11.
+      - archlinux # Since GoReleaser v1.13.
```

Comment:
I think this would be better in a follow-up issue.

Reply:
nix does not need extra support, as far as I can tell - they host a definition in nixpkgs: https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/tools/misc/editorconfig-checker/default.nix

Reply:
> I think this would be better in a follow-up issue.

Depends on what you mean by "follow-up issue", if you mean, it can be done after this PR has been merged, yes, sure why not.

But if you mean, we can release v3.1.0, without publising on Arch Linux and Nix registries, I strongly disagree.

(I was thinking, that this PR was the release cut-off, for v3.1.0, and directly after it has been merged, we release it, but we can make futher PRs after this one before releasing v3.1.0, sure)

Reply:
@mstruebing Could you please, share information, on how it's currently done to publish on Arch Linux and Nix registries, what is needed in terms of keys/secrets, and if you have ideas to automate it?

Reply:
> Depends on what you mean by "follow-up issue", if you mean, it can be done after this PR has been merged, yes, sure why not.

I meant as an actual issue outside of this PR.

> But if you mean, we can release v3.1.0, without publishing on Arch Linux and Nix registries, I strongly disagree.

From having read the makefile and the existing automation, and my general experiences with distro packaging, I can tell that we have no existing upload to either of Nix and Arch. Typically we do a release, and the maintainers of these packages (none of them are people I recognize from the organization members) will pull our source and build the package according to their build instructions (where Nix has their own build instructions, and I have no idea how Arch stores their build instructions). So technically speaking: we cannot publish for Arch and Nix at all - their respective maintainers will need to pull.

> (I was thinking, that this PR was the release cut-off, for v3.1.0, and directly after it has been merged, we release it, but we can make further PRs after this one before releasing v3.1.0, sure)

It's more like with this PR merged we get the release-PR created by release-please and once that PR gets merged the actual release will happen.